src/modules/assessments/components/AssessmentForm.tsx

```diff
@@ -165,54 +169,64 @@ const AssessmentForm = ({
 
   const canEdit = enrollment?.access.canEditEnrollments;
 
+  usePrintTrigger({
+    startReady: isPrintView,
+    hold: !enrollment,
+    timeout: 3000,
```

Comment:
How about instead of the timeout (or in addition to a very brief timeout), we preload of all the picklists? That way, each DynamicField is just retrieving the picklist form the cache. (we'd need to support configuring the fetchPolicy used in DynamicField).

I think the timeout was a good call for timeboxing, but I like the idea of "frontloading" the loading state for the pick lists anyway, because it will help us get rid of that glitching thing Prior Living Situation where it takes a moment for the conditional logic to be applied while the picklist is fetched.


```
// DynamicFormFields.tsx

const { loading } = usePreloadPicklists(itemMap)
if (loading) return loading;

render the stuff
```


```
// usePreloadPicklists.tsx

const pickListTypes = Object.values(itemMap).map(item => item.pickListReference).filter(reference => reference && Object.values<string>(PickListType).includes(reference))

// Run useGetPickListQuery once per pick list type (?), which would get queried in a batch 

return { loading: anyQueriesLoading }
```

Reply:
@gigxz That sounds ideal to me. Would you like me to do that as part of this work, or do you want to split that into a separate ticket?

Reply:
Cool. Let's get this merged; I'll make a separate ticket for it and assign it to you. Feel free to do it if you have any extra time for HMIS this sprint, if not no worries, we can plan for it next week. Thanks!!